File: src/main/java/nl/asdproject2/GreetingServer.java
package nl.asdproject2;

import java.net.*;
import java.io.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Observable;
import java.util.Observer;

public class GreetingServer extends Observable{
    private ServerSocket serverSocket;
    private static List<Observer> clients;
    private static GreetingServer greetingServer;

    public GreetingServer(int port) throws IOException {
        serverSocket = new ServerSocket(port);
        clients = new ArrayList<>();
        greetingServer = this;
        run();
    }

    public void run() {
        while(true) {
            try {
                Socket server = serverSocket.accept();
                System.out.println("Just connected to " + server.getRemoteSocketAddress());

                InputReaderThread client = new InputReaderThread(server.getInputStream(), server.getOutputStream());
                clients.add(client);
                client.start();
            } catch (IOException e) {
                e.printStackTrace();
                break;
            }
        }
    }

    public static void main(String [] args) {
        int port = 6789;
        try {
            new GreetingServer(port);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void update(String message)
    {
        for (Observer client : clients)
        {
            client.update(greetingServer, message);
        }
    }
}

File: src/main/java/nl/asdproject2/kotlin/GreetingServer.kt
package nl.asdproject2.kotlin

import java.net.*
import java.io.*
import java.util.ArrayList
import java.util.Observable
import java.util.Observer

class GreetingServer @Throws(IOException::class)
constructor(port: Int) : Observable() {
    private val serverSocket: ServerSocket

    init {
        serverSocket = ServerSocket(port)
        clients = ArrayList()
        greetingServer = this
        run()
    }

    fun run() {
        while (true) {
            try {
                val server = serverSocket.accept()
                println("Just connected to " + server.remoteSocketAddress)

                val client = InputReaderThread(server.getInputStream(), server.getOutputStream())
                clients.add(client)
                client.start()
            } catch (e: IOException) {
                e.printStackTrace()
                break
            }

        }
    }

    companion object {
        private lateinit var clients: MutableList<Observer>
        private lateinit var greetingServer: GreetingServer

        @JvmStatic
        fun main(args: Array<String>) {
            val port = 6789
            try {
                GreetingServer(port)
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }

        fun update(message: String) {
            for (client in clients) {
                client.update(greetingServer, message)
            }
        }
    }
}